examples/pub-sub-presence/react/src/app/page.tsx

```diff
@@ -0,0 +1,75 @@
+'use client'
+
+import { useEffect, useState } from 'react';
+import { useAbly, usePresence, usePresenceListener } from 'ably/react';
+
+interface Viewer {
+  clientId: string;
+  isOnline: boolean;
+}
+
+export default function Home() {
+  const [viewers, setViewers] = useState<Viewer[]>([]);
+
+  const ably = useAbly();
+  const currentClientId = ably?.auth.clientId;
+
+  const { presenceData } = usePresenceListener('viewer-presence', (presence) => {
+    switch (presence.action) {
```

Comment:
no need to do this, `presenceData` is always represent actual presence set, so it can be replaced with:

```ts
const { presenceData } = usePresenceListener('viewer-presence')
const viewers: Viewer[] = useMemo(() => presenceData.map(member => ({
      clientId: member.clientId,
      isOnline: true
},  presenceData);

```